examples/common/mod.rs

```diff
@@ -16,3 +17,67 @@ pub enum Rpc {
     /// Starts the connectivity and data exchange test between us and given QuicP2p peers.
     StartTest(Vec<Peer>),
 }
+
+/// Receivers side for events
+pub struct EventReceivers {
+    pub node_rx: mpmc::Receiver<Event>,
+    pub client_rx: mpmc::Receiver<Event>,
+}
+
+impl EventReceivers {
+    #[allow(unused)]
+    pub fn recv(&self) -> Result<Event, mpmc::RecvError> {
+        let mut sel = mpmc::Select::new();
+        let client_idx = sel.recv(&self.client_rx);
+        let node_idx = sel.recv(&self.node_rx);
+        let selected_operation = sel.ready();
+
+        if selected_operation == client_idx {
+            self.client_rx.recv()
+        } else if selected_operation == node_idx {
+            self.node_rx.recv()
+        } else {
+            panic!("invalid operation");
+        }
```

Comment:
I think this would be simpler using the [select](https://docs.rs/crossbeam-channel/0.4.0/crossbeam_channel/macro.select.html) macro.

Reply:
The select macro do not return the error. Also manual select is how we do it in vault because the source are separated. Apply to all of these comments.

Reply:
> The select macro do not return the error.

I think it does?

```rust
select! {
    recv(self.client_rx) -> ev => /* ev is Result */,
    recv(self.node_rx) -> ev => /* ev is Result */,
}
```

> Also manual select is how we do it in vault because the source are separate

We do it like that because the set of channels to select over isn't statically known. But that's not the case here. `select!` would work just fine and make things simpler.

Reply:
That said, if you don't want to do this due to lack of time or whatever then I'm OK leaving this to a followup PR.

Reply:
Correct, I'll raise an issue.

Reply:
Created issue https://github.com/maidsafe/quic-p2p/issues/96 and added it to the fleming routing project.